tests/CMakeLists.txt

```diff
@@ -39,8 +39,15 @@ set(DETFUNCTION_TEST_SOURCES
   check_detFunction.c
 )
 
+set(LISTS_TEST_SOURCES
+  check_lists.c
+)
+
 add_executable(check_gammatone ${GAMMATONE_TEST_SOURCES} ${ARRAY_TEST_SOURCES})
 add_executable(check_detFunction ${DETFUNCTION_TEST_SOURCES}
   ${ARRAY_TEST_SOURCES})
+add_executable(check_lists ${LISTS_TEST_SOURCES})
```

Comment:
If I am understanding this correctly, are you making a different executable for each test file? It doesn't need to be changed for this commit, but if that is the case down the line I think it would be better to have a single test executable that runs all tests. Once we start adding lots of tests it will become a real pain otherwise.

Reply:
I hadn't given this a ton of thought and was just following the simple tutorial that they provide.

The basic idea in the check framework is that you organize test cases into suites and you might have a few suites in a file. I think the idea is for test suites to consolidate test cases that have common setup and teardown steps. I generally agree that having a ton of files wouldn't be great. There are 2 tradeoffs to having a single file executable:

1. We need to deal with header files (this isn't a problem)
2. It becomes harder to selectively run a set of tests. It is doable to [selectively run a test suite](https://libcheck.github.io/check/doc/check_html/check_4.html#Selecting-Tests-by-Suite-or-Test-Case). This probably isn't much of an issue for now (since it doesn't take much time to run everything). However in the future, we might want to be careful how we group together test suites (e.g. we have more than one test suite for gammatone filters that each hold test cases that test different aspects of the implementation). 

To summarize: I agree that it would be useful to have a single executable, but we should be somewhat mindful that if we have a lot of tests, we will need to be able to intelligently specify subsets of tests to run.